simulation/behavior_tree_plugin/include/behavior_tree_plugin/action_node.hpp

```diff
@@ -84,41 +83,42 @@ class ActionNode : public BT::ActionNodeBase
       BT::InputPort<lanelet::Ids>("route_lanelets"),
       BT::InputPort<std::optional<double>>("target_speed"),
       BT::InputPort<std::shared_ptr<hdmap_utils::HdMapUtils>>("hdmap_utils"),
-      BT::InputPort<std::shared_ptr<traffic_simulator::CanonicalizedEntityStatus>>("entity_status"),
+      BT::InputPort<std::shared_ptr<traffic_simulator::CanonicalizedEntityStatus>>("canonicalized_entity_status"),
       BT::InputPort<std::shared_ptr<traffic_simulator::TrafficLightManager>>("traffic_light_manager"),
       BT::InputPort<traffic_simulator::behavior::Request>("request"),
       BT::OutputPort<std::optional<traffic_simulator_msgs::msg::Obstacle>>("obstacle"),
-      BT::OutputPort<std::shared_ptr<traffic_simulator::CanonicalizedEntityStatus>>("updated_status"),
       BT::OutputPort<traffic_simulator_msgs::msg::WaypointsArray>("waypoints"),
       BT::OutputPort<traffic_simulator::behavior::Request>("request"),
       // clang-format on
     };
   }
   auto getBlackBoardValues() -> void;
   auto getEntityStatus(const std::string & target_name) const
-    -> traffic_simulator::CanonicalizedEntityStatus;
+    -> const traffic_simulator::CanonicalizedEntityStatus &;
   auto getDistanceToTargetEntityPolygon(
     const math::geometry::CatmullRomSplineInterface & spline, const std::string target_name,
     double width_extension_right = 0.0, double width_extension_left = 0.0,
     double length_extension_front = 0.0, double length_extension_rear = 0.0) const
     -> std::optional<double>;
+
+  auto setCanonicalizedEntityStatus(const traffic_simulator::EntityStatus & entity_status) -> void;
   auto calculateUpdatedEntityStatus(
     double target_speed, const traffic_simulator_msgs::msg::DynamicConstraints &) const
-    -> traffic_simulator::CanonicalizedEntityStatus;
+    -> traffic_simulator::EntityStatus;
   auto calculateUpdatedEntityStatusInWorldFrame(
     double target_speed, const traffic_simulator_msgs::msg::DynamicConstraints &) const
-    -> traffic_simulator::CanonicalizedEntityStatus;
+    -> traffic_simulator::EntityStatus;
 
 protected:
   traffic_simulator::behavior::Request request;
   std::shared_ptr<hdmap_utils::HdMapUtils> hdmap_utils;
   std::shared_ptr<traffic_simulator::TrafficLightManager> traffic_light_manager;
-  std::shared_ptr<traffic_simulator::CanonicalizedEntityStatus> entity_status;
+  std::shared_ptr<traffic_simulator::CanonicalizedEntityStatus> canonicalized_entity_status;
   double current_time;
   double step_time;
   double default_matching_distance_for_lanelet_pose_calculation;
   std::optional<double> target_speed;
-  std::shared_ptr<traffic_simulator::CanonicalizedEntityStatus> updated_status;
+  std::shared_ptr<traffic_simulator::EntityStatus> non_canonicalized_updated_status;
```

Comment:
I analysed the code and this variable does not appear to be used.

Reply:
@hakuturu583 
You're right, since the [alternative solution](https://github.com/tier4/scenario_simulator_v2/pull/1320#discussion_r1695775781) was introduced it is not used, it's a souvenir - I'm already removing it. :bow: